data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarderTest.java

```diff
@@ -43,11 +43,18 @@
 
 @ExtendWith(MockitoExtension.class)
 class RemotePeerForwarderTest {
+    private static final int TEST_BUFFER_CAPACITY = 3;
+    private static final int TEST_BATCH_SIZE = 3;
+    private static final int TEST_TIMEOUT_IN_MILLIS = 500;
+    private static final PeerForwarderReceiveBuffer<Record<Event>> peerForwarderReceiveBuffer =
+            new PeerForwarderReceiveBuffer<>(TEST_BUFFER_CAPACITY, TEST_BATCH_SIZE);
```

Comment:
This way, all created peer forwarders under test share the same receive buffer. Should we move it inside `createObjectUnderTest` method and create new buffer for each peer forwarder instead?